### ISSUE TITLE
Locating `UIViewRepresentable` views when multiple views of the same type exists

### ISSUE THREAD

**Issue by Nillerr** [nalexn/ViewInspector#166]
Hi @nalexn 👋 

Thanks for the amazing work you've put into ViewInspector!

We use a lot of our in-house custom iOS components, where some are built on top of UIKit and thus wrapped in `UIViewRepresentable`s. We have noticed an issue in trying to locate these elements by `accessibilityIdentifier()` when multiple instances of one of these elements are contained in a single `VStack`:

```swift
struct UIKitView: UIViewRepresentable {
    ...
}

struct WrapperView: View {
    var body: some View {
        UIKitView(...)
    }
}

struct ContentView: View {
    var body: some View {
        VStack {
            WrapperView(...)
                .accessibilityIdentifier("view-1")

            WrapperView(...)
                .accessibilityIdentifier("view-2")
        }
    }
}
```

Using a structure such as this, it appears to be impossible to select the `actualView().uiView()` with the identifier `view-2`, as calling those functions always returns the `UIView` returned by the `UIKitView` within `view-1`.

Looking at the source code for`ViewHosting`, I see this extension through the call path of `uiView()`:

```swift
    static func lookup<V>(_ view: V.Type) throws -> V.UIViewType
        where V: Inspectable & UIViewRepresentable {
            let name = Inspector.typeName(type: view)
            let viewHost = window.descendant(nameTraits: ["ViewHost", name])
            guard let view = viewHost?.subviews.compactMap({ $0 as? V.UIViewType }).first else {
                throw InspectionError.viewNotFound(parent: name)
            }
            return view
    }
```

Which appears to (in combination with `window.descendant(nameTraits:)`) select the _first_ view of the given type it locates.

Admittedly, I haven't investigated how UIKit and SwiftUI interacts on these low levels, so I'm hoping on your expertise to shed some light on why it appears to always only find the first view of the given type, and if it would be possible to resolve this issue 🙏

**Comment by nalexn** [comment]
Yeah, I can see the problem, the code that looks up the UIView does not handle this case properly. I'll explore how this can be fixed.
